crates/node-optimism/src/engine.rs

```diff
@@ -15,6 +18,9 @@ impl EngineTypes for OptimismEngineTypes {
     type PayloadAttributes = OptimismPayloadAttributes;
     type PayloadBuilderAttributes = OptimismPayloadBuilderAttributes;
     type BuiltPayload = EthBuiltPayload;
+    type ExecutionPayloadV1 = ExecutionPayloadV1;
+    type ExecutionPayloadV2 = ExecutionPayloadEnvelopeV2;
+    type ExecutionPayloadV3 = ExecutionPayloadEnvelopeV3;
```

Comment:
once merged we can add a dedicated OP V3 payload type in alloy because this is now diverged from ethereum

and I think we also want to change the `BuiltPayload` type to a new `OptimismBuiltPayload` that then should include the ChainSpec so we no longer need the additional check `parent_beacon_block_root` check